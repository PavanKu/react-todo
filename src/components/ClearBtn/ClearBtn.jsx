import React from "react";
import PropTypes from "prop-types";

import StyledBtn from "./ClearBtn.styled";

export const DEFAULT_TEXT = "Clear completed";
export const DEFAULT_SHOW = true;

function ClearBtn({ handleClear, label = DEFAULT_TEXT, show = DEFAULT_SHOW }) {
  const handleClick = () => {
    if (show && handleClear) {
      handleClear();
    }
  };

  return (show && <StyledBtn onClick={handleClick}>{label}</StyledBtn>) || null;
}

ClearBtn.propTypes = {
  handleClear: PropTypes.func,
  label: PropTypes.string,
  show: PropTypes.bool,
};
export default ClearBtn;
