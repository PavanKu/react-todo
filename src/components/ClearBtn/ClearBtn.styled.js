import styled from "styled-components";

const StyledBtn = styled.button`
  font-size: 14px;
  color: #777;
  cursor: pointer;
  outline: none;
  border: none;
  background: none;
  :hover {
    text-decoration: underline;
    text-decoration-color: #777;
  }
`;

export default StyledBtn;
