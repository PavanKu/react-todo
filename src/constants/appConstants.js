const TAB_OPTIONS = {
  ALL: "All",
  ACTIVE: "Active",
  COMPLETED: "Completed",
};

export { TAB_OPTIONS };
