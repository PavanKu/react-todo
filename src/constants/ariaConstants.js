const ARIA_ROLE_HEADING = "heading";
const ARIA_ROLE_TEXTBOX = "textbox";
const ARIA_ROLE_CHECKBOX = "checkbox";
const ARIA_ROLE_BUTTON = "button";
const ARIA_ROLE_TABLIST = "tablist";
const ARIA_ROLE_TAB = "tab";

const ARIA_TAB_SELECTED_ATTRIBUTE = "aria-selected";

export {
  ARIA_ROLE_HEADING,
  ARIA_ROLE_TEXTBOX,
  ARIA_ROLE_CHECKBOX,
  ARIA_ROLE_BUTTON,
  ARIA_ROLE_TABLIST,
  ARIA_ROLE_TAB,
  ARIA_TAB_SELECTED_ATTRIBUTE,
};
