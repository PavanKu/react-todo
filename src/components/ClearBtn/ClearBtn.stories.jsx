import React from "react";
import { storiesOf } from "@storybook/react";
import { withKnobs, boolean, text } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";

import ClearBtn from "./ClearBtn";

const DefaultClearBtn = () => <ClearBtn />;
const ClearBtnWithCustomLabel = () => <ClearBtn label="Click me" />;
const Playground = () => (
  <ClearBtn
    show={boolean("show", true)}
    label={text("label", "Hire me")}
    handleClear={action("ClearBtn:click:")}
  />
);

storiesOf("Clear Button", module)
  .addDecorator(withKnobs)
  .add("Default Button", DefaultClearBtn)
  .add("Custom Label", ClearBtnWithCustomLabel)
  .add("Playground", Playground);
