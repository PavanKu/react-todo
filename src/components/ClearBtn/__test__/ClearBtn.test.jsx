import React from "react";
import { cleanup, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import ClearBtn, { DEFAULT_TEXT } from "../ClearBtn";
import { ARIA_ROLE_BUTTON } from "../../../constants/ariaConstants";

describe("<ClearBtn />", () => {
  afterEach(cleanup);

  it("should render without crash", () => {
    render(<ClearBtn />);
  });

  it("should render with default props", () => {
    const { getByRole } = render(<ClearBtn />);
    const elementByRole = getByRole(ARIA_ROLE_BUTTON);

    expect(elementByRole).toHaveTextContent(DEFAULT_TEXT);
  });

  it("should render with custom label", () => {
    const sampleLabel = "Click me";
    const { getByRole } = render(<ClearBtn label={sampleLabel} />);
    const elementByRole = getByRole(ARIA_ROLE_BUTTON);

    expect(elementByRole).toHaveTextContent(sampleLabel);
  });

  it("should hide button based on props", () => {
    const { container } = render(<ClearBtn show={false} />);
    const btnElement = container.querySelector("button");

    expect(btnElement).toBeNull();
  });

  it("should invoke handler", () => {
    let handlerInvoked = false;
    const { getByRole } = render(
      <ClearBtn handleClear={() => (handlerInvoked = true)} />
    );
    const elementByRole = getByRole(ARIA_ROLE_BUTTON);

    userEvent.click(elementByRole);

    expect(handlerInvoked).toStrictEqual(true);
  });

  it("should match snapshot", () => {
    const { container } = render(<ClearBtn show label={"Click me"} />);
    expect(container).toMatchSnapshot();
  });
});
