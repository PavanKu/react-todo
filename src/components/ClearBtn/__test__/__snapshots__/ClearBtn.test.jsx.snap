// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClearBtn /> should match snapshot 1`] = `
<div>
  <button
    class="sc-AxjAm hnSQbp"
  >
    Click me
  </button>
</div>
`;
